### ISSUE TITLE
CUE   multiple list merge failed 

### ISSUE THREAD

**Issue by wxuedong** [kubevela/kubevela#3483]
vela version
    Version: 1.1.6
    GitRevision: git-844f479f
    GolangVersion: go1.17.2

define a component  parameter:
```
parameter: {
  ports: *[8088] | [...int]
  programParam: *"-Dargname=argvalue" | string
  processParam: *"-Dfile.encoding=UTF-8" | string
}
#PortEnv: [
  for i,v in parameter.ports {
            if i  == 0 {
            {
	            "name": "SERVICE_PORT",
	            "value":   "\(v)"
            },
            }
            if i != 0 {
	            {
	            "name": "SERVICE_PORT_"+"\(i)"+"",
	            "value":   "\(v)"
	            },
            }
    }
  ]
#Process_Env: [
	{
	"name": "PROCESS_PARAM",
	"value": parameter.processParam
	},
	{
	"name": "PROGRAM_PARAM",
	"value": parameter.programParam
},
 ]
```
I want to merge two list  together，error executing the following code

```
#Env:  [
	for k, v in #PortEnv {
			name: v.name
			value: v.value
	},
	for k, v in #Process_Env {
		 	  name: v.name
			value: v.value
	},
  ]
```
error message :

`Error: failed to parse CUE: multiple comprehensions per list not yet supported (and 1 more errors)`

The same method is ok in the official CUE Playground of Cuelang

<img width="1503" alt="image" src="https://user-images.githubusercontent.com/15843627/159227886-15d6cc5e-d21e-40ef-b102-8b3aaf3463c1.png">

```
#PORTENV: [
{key:"test",value:"eee"}
]
#PORTENV1: [
{key:"test22",value:"eee22"}
]
env:[
   for k, v in #PORTENV {
        envname: v.key
        envvalue: v.value
    },
   for k, v in #PORTENV1 {
          envname: v.key
        envvalue: v.value
    },
]
```

**Comment by zxbyoyoyo** [comment]
cuelang v0.2.2 does not support this, you can wait the pr to be merged #3478

**Comment by wonderflow** [comment]
CUE 0.4 is merged in master of KubeVela, please give it a try! https://github.com/kubevela/kubevela/pull/4425
